tools/codegen/index.js

```diff
@@ -110,47 +127,128 @@ class DeviceModel {
       }
 }
 
-// extract schema name from input filename
-const deviceName = path.parse(options.deviceModel).name.split('.')[0];
-console.log(`Validating device model '${deviceName}' from file '${options.deviceModel}' against schema file '${options.schema}'...`);
-
-// try {
-//     // Determining if the file is yaml or json and parsing accordingly.
-//     const data = (() => {
-//         const extension = path.extname(options.deviceModel);
-//         if (extension === ".yaml" || extension === ".yml") {
-//             return yaml.parse(fs.readFileSync(options.deviceModel, 'utf8'));
-//         } else { // Default
-//             return JSON.parse(fs.readFileSync(options.deviceModel));
-//         }
-//     })();
-//     if (validator.validateDevice(data)) {
-//         const CodeGen =  require(`./${options.language}/${options.language}gen.js`);
-//         const dm = new DeviceModel(options.deviceModel, validator, data);
-//         const codeGen = new CodeGen(dm, options.output);
-//         codeGen.generate();
-//     }
-
-// } catch (why) {
-//     console.log(why.message);
-//     process.exit(typeof why.error === 'number' ? why.error : 1);
-// }
+const deviceName = path.parse(options.deviceModel).name.split('.')[0]; //Extracts the name prefix (eg device.minimal.json -> device)
+console.log(`Validating device model '${deviceName}' from file '${options.deviceModel}' against schema file '${options.schema}'...`); //Begins validation
```

Comment:
```suggestion
//Extracts the name prefix (eg device.minimal.json -> device)
const deviceName = path.parse(options.deviceModel).name.split('.')[0];
//Begins validation
console.log(`Validating device model '${deviceName}' from file '${options.deviceModel}' against schema file '${options.schema}'...`);
```
I prefer comments like this on their own lines so that it is obvious the code hasn't actually changed.